specs/required-inputs/README.md

```diff
@@ -43,17 +47,36 @@ This will be implemented using a new `nimble-icon-asterisk` icon ([named consist
     <label part="label" class="label">
         <slot></slot>
     </label>
-    ${when(x => x.required, html`
+    ${when(x => x.requiredVisible, html`
         <nimble-icon-asterisk severity="error"></nimble-icon-asterisk>
     `)}
 </div>
 ```
 
-Note that for text-field, text-area, and number-field, we will also remove the visual if the control has the `readonly` attribute set, since [validation is not enforced on a readonly input](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#attribute_interactions). This doesn't apply to select, combobox, or radio buttons, because they do not support `readonly`.
+Note that for text-field, text-area, and number-field, the `readonly` attribute will prevent the display of the asterisk icon, since [validation is not enforced on a readonly input](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#attribute_interactions). This doesn't apply to select, combobox, or radio buttons, because they do not support `readonly`.
 
 Styles will be shared.
 
-### Form validation
+
+### Angular & Blazor
+
+The `required-visible` attribute will be added to the wrappers.
+
+## Documentation
+
+We will document the new `required-visible` attribute, but we will not add comprehensive documentation about form validation or the `required` attribute.
+
+## Testing
+
+Visual matrix test cases will be added for the affected components.
+
+## Alternative Implementations / Designs
+
+None
+
+## Form validation
+
+We are opting to keep our API separate from the API that actually triggers form validation, but for completeness, an overview of Nimble's support of the `required` attribute follows.
 
 #### Vanilla HTML
```

Comment:
A comment that I made yesterday was that the `required-visual` state has a visual for sighted users that a field is required and that it should also map to the aria-required stated for aria users to get the same information that a field is required. The discussion below seems to indicate differently and primarily discusses association via the `required` attribute still. Curious how we resolve that.

Reply:
You may have said this yesterday and I just forgot, but do you think setting `error-visible` should also set `aria-invalid`? (I would think the answer is no.) My takeaway from our conversation was that we want to decouple control of the visuals from the semantics they represent. This is supported by the fact that the attributes are named `foo-visible`, which implies nothing about functionality/semantics, but only a visual cue. Accessibility of an element's required-ness is coupled to the functional attribute, i.e. `required` (or `aria-required`, which would be bound to the presence of `required` on a host element). I don't see how we can couple the presence of our visual indicator (asterisk) with the accessibile indicator (i.e. screen reader announcing "required") unless we are also coupled to the functional attribute (`required`).

Reply:
> Accessibility of an element's required-ness is coupled to the functional attribute, i.e. required

I don't believe that is the right mental model. The whole point of aria is being able to get the right aria to arbitrary controls, i.e. making a `<div>` aria correct does not involve the `required` attribute, it involves a handful of `aria-` attributes.

Reply:
Chatted offline and discussed the following:

The semantics of the `required` attribute include:
- behavior (form-association)
- presentation (visuals and aria)

In a very ideal world:
- The `required` attribute and others associated with validation, i.e. `min`, `max` would handle both behavior and presentation. 
- Frameworks would rely on native form association and not implement parallel form validity systems

Today as a compromise for frameworks:
- The `required` attribute and others associated with validation, i.e. `min`, `max` would handle just behavior
- The user handles presentation via `error-visible` and `required-visible` (and as a reminder presentation includes visuals and aria)
- An application (and ideally a pattern for consistency across applications) should determine how aligned we are with the semantics of `required`, i,e, when `required` + `disabled` should the presentation be required or not-required
- Frameworks like angular will set the `required` attribute but opt out of the behavior as forms will be marked `novalidate` due to Angular's design. The user will still need to opt-in to presentation via `required-visible`.

Today what will actually happen because of implementation limitations:
- We do not use Form-Association apis directly and rely on FAST's polyfills for forma association
- The polyfill for Form-Association will forward required attributes to internal input elements used by the polyfill which will surface to aria kinda outside our control.
- So `required` may or may not influence presentation aria (due to limitations of the polyfill design) but will not influence presentation visuals. But `required-visible` will influence presentation of both visual and aria.

Reply:
I have tried to add the most important points related to our reasoning to the HLD.